firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/common/CommonUtilsTest.java

```diff
@@ -288,6 +295,71 @@ public void testResolveBuildId_bothIds() throws Exception {
         });
   }
 
+  public void testCapFileCount() throws Exception {
+    final Context context = getContext();
+
+    final File testDir = new File(context.getFilesDir(), "trim_test");
+    testDir.mkdirs();
+    // make sure the directory is empty. Doesn't recurse into subdirs, but that's OK since
+    // we're only using this directory for this test and we won't create any subdirs.
+    for (File f : testDir.listFiles()) {
+      if (f.isFile()) {
+        f.delete();
+      }
+    }
+
+    final int maxFiles = 4;
+    // create a bunch of non-cls files that don't match the capFileCount
+    for (int i = 0; i < maxFiles + 2; ++i) {
+      new File(testDir, "whatever" + i + ".blah").createNewFile();
+    }
+    assertEquals(maxFiles + 2, testDir.listFiles().length);
+    final FilenameFilter clsFilter = CrashlyticsController.SESSION_FILE_FILTER;
+    // This should have no effect - nothing matches the filter yet
+    Utils.capFileCount(
+        testDir, clsFilter, maxFiles, CrashlyticsController.SMALLEST_FILE_NAME_FIRST);
+    assertEquals(maxFiles + 2, testDir.listFiles().length);
+
+    // create two groups of empty crash files. The first set will be deleted after the 2nd
+    // is created and trim is invoked.
+    final Set<File> oldFiles = new HashSet<File>();
+    for (int i = 0; i < maxFiles; ++i) {
+      oldFiles.add(createEmptyClsFile(testDir));
+
+      // This should have no effect - we're not over the limit
+      Utils.capFileCount(
+          testDir, clsFilter, maxFiles, CrashlyticsController.SMALLEST_FILE_NAME_FIRST);
+      assertEquals(oldFiles.size(), testDir.listFiles(clsFilter).length);
+    }
+    // The filesystem only has 1sec precision, so we need to sleep long enough that
+    // the timestamps in the 2nd set of files are later than the ones in the first.
+    Thread.sleep(1500);
```

Comment:
Thread.sleep behaviors are not ideal in tests because it can slow your overall test suite. Are there any alternative ways we could do this (e.g., abstract / mock more systems, or loop and check for state existing within some time bounds, and exit early if state becomes present?